src/main.rs

```diff
@@ -49,12 +50,19 @@ fn main() -> anyhow::Result<()> {
             let cli_error = match error.downcast::<CLIError>() {
                 Ok(cli_error) => cli_error,
                 Err(error) => {
-                    let sources = error
-                        .source()
-                        .map(|error| vec![CLIError::new(error.to_string().as_str())])
-                        .unwrap_or_default();
+                    // Convert other errors to CLIError
+                    let cli_error = match error.downcast::<ValidationError<String>>() {
+                        Ok(validation_error) => CLIError::from(validation_error),
+                        Err(error) => {
+                            let sources = error
+                                .source()
+                                .map(|error| vec![CLIError::new(error.to_string().as_str())])
+                                .unwrap_or_default();
 
-                    CLIError::new(&error.to_string()).caused_by(sources)
+                            CLIError::new(&error.to_string()).caused_by(sources)
+                        }
+                    };
+                    cli_error
```

Comment:
@coderabbitai We could add simple unit tests for error downcasting. Write a function that takes the error that's produced from run_blocking, and checks various downcasting branches. Add mod tests to write some unit tests for those conversions.

Reply:
@coderabbitai suggest actual tests not just comments.

Reply:
@coderabbitai Can you write simulate_errors function & some sample cases ?